### PR TITLE
feat: add tasklist to support test standalone camunda 

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

- Move Standalone Camunda tests/util classes to Zeebe Root
- Include tasklist on TestStandaloneCamunda classes 

## Related issues

closes #https://github.com/camunda/camunda/issues/19962

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -82,9 +82,6 @@
           <value>!true</value>
         </property>
       </activation>
-      <modules>
-        <module>qa</module>
-      </modules>
     </profile>
```
Comment (Zelldon) [inline pom.xml:85]: ❌ ❓ Why was this removed? This is necessary to execute the QA module.

File: zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestCamundaApplication.java [hunk 2]
```diff
@@ -5,7 +5,7 @@
  * Licensed under the Camunda License 1.0. You may not use this file
  * except in compliance with the Camunda License 1.0.
  */
-package io.camunda.qa.util.cluster;
+package io.camunda.zeebe.qa.util.cluster;
```
Comment (Zelldon) [inline zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestCamundaApplication.java:8]: ❌ ❓ Sorry maybe I missed something, but why you're moving the classes back to the Zeebe qa modules?

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
I haven't completed the review, but I think we should first clarify what you want to achieve here. 

I'm wondering why you moved the classes back to the Zeebe/qa modules, as they have been recently moved out to the root `qa/integration-tests` in order to cover tests that are related to the whole platform (e.g. StandaloneApplication, etc.) 

Lets first clarify and discuss this, before we move on here.

File: pom.xml (hunk 1 above)
Comment (nathansandi) [reply pom.xml:85]: This `qa` module was not used at the end, for some reason only TestStandaloneCamunda was there, but was not being reading correctly by the build. I was looking the classes and the clustering tests was placed in the zeebe folder (following the second comment).

Did I miss something on this changes @Zelldon ?

File: zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestCamundaApplication.java (hunk 2 above)
Comment (nathansandi) [reply zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestCamundaApplication.java:8]: It was not clear which folders it should be, because the qa module on the root was not functional at all, so the reason I moved the to the folder that was funcitonal, I guess was a missunderstanding for my part

Because half of tests (reading by the CI) was on Zeebe/zeebe, and only the Standalone Camunda App was on the root, but not correctly read by intellij

In order to clarify, which folder should has this? So I can move it and keep all on just one, or should we have all the tests over the root?  (It is not really clear on my side)

Comment (Zelldon) [reply zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestCamundaApplication.java:8]: It would be good for the next time if we could clarify this upfront :) so please if in doubt ask.

The qa on root level was added via https://github.com/camunda/camunda/pull/19757 and mentioned in several updates in slack https://camunda.slack.com/archives/C06HTSPD5AP/p1719472381357659


>  because the qa module on the root was not functional at all

Well that is not true. Tests are executed on CI, you can see for example on recent main change https://github.com/camunda/camunda/actions/runs/9776526029/job/26989329015

[An successful run](https://github.com/camunda/camunda/actions/runs/9776526029/job/26989329015#step:9:118)

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.camunda.it.StandaloneCamundaTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 55.24 s -- in io.camunda.it.StandaloneCamundaTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by Zelldon** [comment]
As I described in my [comment here](https://github.com/camunda/camunda/pull/19961#discussion_r1664030501) the QA were working, and in the correct place. 

Please revert the changes you did regarding this.

Afterwards, what is necessary from your side is to extend the TestStandaloneCamunda, such that the right modules are loaded, similar to what you did. Then you can add an additional test in `StandaloneCamundaTest` that:

**Comment by nathansandi** [comment]
Hey @Zelldon 
Thanks for the comments, for some reason the problem was on my maven + intellij in order to sync the files, and I had to change some setup on my intellij, now its working fine, and I extended the TestStandaloneCamunda to start the Tasklist profile

Note I havent add Tasklist Integration Tests yet, because I will add as a follow-up of Search User Task, and will focus to write this tests in a separate PR using the V2 endpoints. And with that I will be able write the tests using the client in order to:
- Retrieve Task Created
- Assign/Unassign
- Complete

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 3]
```diff
@@ -261,13 +236,6 @@ public TestStandaloneCamunda withExporter(final String id, final Consumer<Export
     return this;
   }
 
-  /**
-   * Sets the broker's working directory, aka its data directory. If a path is given, the broker
-   * will not delete it on shutdown.
-   *
-   * @param directory path to the broker's root data directory
-   * @return itself for chaining
-   */
```
Comment (Zelldon) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:270]: ❌ Can you explain why you removed all comments in this class?

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 4]
```diff
@@ -69,37 +64,41 @@ public final class TestStandaloneCamunda extends TestSpringApplication<TestStand
...
 
   public TestStandaloneCamunda() {
     super(
-        BrokerModuleConfiguration.class,
+        CommonsModuleConfiguration.class,
         OperateModuleConfiguration.class,
+        TasklistModuleConfiguration.class,
         WebappsModuleConfiguration.class,
-        CommonsModuleConfiguration.class,
-        // test overrides - to control data clean up; (and some components are not installed on
```
Comment (Zelldon) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:79]: ❌ I would keep this comment

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 5]
```diff
@@ -69,37 +64,41 @@ public final class TestStandaloneCamunda extends TestSpringApplication<TestStand
...
-        // Tests)
-        TestElasticsearchSchemaManager.class,
-        TestSchemaStartup.class);
+        BrokerModuleConfiguration.class,
+        GatewayModuleConfiguration.class,
```
Comment (Zelldon) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:76]: ❓ Was this necessary? 🤔 It was working without before.

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Thanks @nathansandi thanks for updating the PR.

I had some comments on your PR, see below. Furthermore, I think it is worthwhile to already check whether Tasklist is correctly up and running. Did you verified this somehow?

Please also check the CI, it is currently failing, as license headers a missing.



> Note I havent add Tasklist Integration Tests yet, because I will add as a follow-up of Search User Task, and will focus to write this tests in a separate PR using the V2 endpoints. And with that I will be able write the tests using the client in order to:
> 
> - Retrieve Task Created
> - Assign/Unassign
> - Complete

I see. Would be great to use the client then. But this is not done I suppose? Could you already verify something on the v1 API, and replace it later? 🤔 


Please be aware that I'm not available until 05.08. I guess you want to continue with this, so it would probably someone else takes over, like @tmetzke or @houssain-barouni

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java (hunk 5 above)
Comment (nathansandi) [reply qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:76]: The BrokerModule yes, otherwise Tasklist is not able to retrieve the partitions in correct way. 
The GatewayModuleConfiguration not, removed this one.

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java (hunk 4 above)
Comment (nathansandi) [reply qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:79]: Removed by accident, gonna re-add it.

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java (hunk 3 above)
Comment (nathansandi) [reply qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:270]: Removed by accident in a refactor, going to readd it.

**Comment by nathansandi** [comment]
> Thanks @nathansandi thanks for updating the PR.
> 
> I had some comments on your PR, see below. Furthermore, I think it is worthwhile to already check whether Tasklist is correctly up and running. Did you verified this somehow?
> 
> Please also check the CI, it is currently failing, as license headers a missing.
> 
> > Note I havent add Tasklist Integration Tests yet, because I will add as a follow-up of Search User Task, and will focus to write this tests in a separate PR using the V2 endpoints. And with that I will be able write the tests using the client in order to:
> > 
> > * Retrieve Task Created
> > * Assign/Unassign
> > * Complete
> 
> I see. Would be great to use the client then. But this is not done I suppose? Could you already verify something on the v1 API, and replace it later? 🤔
> 
> Please be aware that I'm not available until 05.08. I guess you want to continue with this, so it would probably someone else takes over, like @tmetzke or @houssain-barouni

The setup can be validated over StandaloneCamunda Test, once it will Start Tasklist  over there. The client for Search User Task is over review, and I will be able to write the tests right after merge this PR 

https://github.com/camunda/camunda/pull/19625

I dont see value in add V1 Tests here, once they already exists on Tasklist App, and write the tests for V2 will be unblocked after merge the above PR.

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 6]
```diff
@@ -111,20 +109,20 @@ public TestStandaloneCamunda start() {
...
+    tasklistProperties.getZeebeElasticsearch().setUrl(esURL);
     return super.start();
   }
 
   @Override
   public TestStandaloneCamunda stop() {
-    // clean up ES/OS indices
     LOGGER.info("Stopping standalone camunda test...");
-    ((TestElasticsearchSchemaManager) bean(SchemaManager.class)).deleteSchema();
+    ((TestOperateElasticsearchSchemaManager) bean(SchemaManager.class)).deleteSchema();
```
Comment (houssain-barouni) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:126]: we should delete Tasklist schema as well

File: pom.xml [hunk 7]
```diff
@@ -40,6 +41,20 @@
     <tag>HEAD</tag>
     <url>https://github.com/camunda/camunda</url>
   </scm>
+  <dependencies>
```
Comment (houssain-barouni) [inline pom.xml:44]: These dependencies should be moved to `qa/util/pom.xml` if required there

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 8]
```diff
@@ -69,37 +63,41 @@ public final class TestStandaloneCamunda extends TestSpringApplication<TestStand
           .withEnv("action.destructive_requires_name", "false");
   private final BrokerBasedProperties brokerProperties;
   private final OperateProperties operateProperties;
+  private final TasklistProperties tasklistProperties;
 
   public TestStandaloneCamunda() {
     super(
-        BrokerModuleConfiguration.class,
+        CommonsModuleConfiguration.class,
+        // test overrides - to control data clean up; (and some components are not installed on)
```
Comment (houssain-barouni) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:71]: I think this comment applies to `TestOperateElasticsearchSchemaManager`

**Review by houssain-barouni — COMMENTED** [review]
We will sync later to review what is remaining.

File: pom.xml [hunk 9]
```diff
@@ -32,6 +32,7 @@
     <module>testing/camunda-process-test-java</module>
     <module>testing/camunda-process-test-spring</module>
     <module>clients/java</module>
+    <module>qa</module>
```
Comment (houssain-barouni) [inline pom.xml:35]: I forgot to tell you to remove this
this should be controlled with `skipQaBuild` property (see `integration-tests-build` below)

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 10]
```diff
@@ -1,21 +1,17 @@
-/*
```
Comment (houssain-barouni) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:1]: you need to put back the license header, the linter will complain about it

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 11]
```diff
@@ -213,28 +210,15 @@ public ZeebeClientBuilder newClientBuilder() {
     return TestGateway.super.newClientBuilder();
   }
 
-  /** Returns the broker configuration */
   public BrokerBasedProperties brokerConfig() {
     return brokerProperties;
   }
 
-  /**
```
Comment (houssain-barouni) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:221]: can you put back all the code comments

**Comment by nathansandi** [comment]
Hey @houssain-barouni  thanks for check this :) 
I have just re-add the comments, fix the license and dependencies. I guess now its good to go

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java [hunk 12]
```diff
@@ -116,15 +127,17 @@ public TestStandaloneCamunda start() {
...
     return super.start();
   }
 
   @Override
   public TestStandaloneCamunda stop() {
     // clean up ES/OS indices
     LOGGER.info("Stopping standalone camunda test...");
-    ((TestElasticsearchSchemaManager) bean(SchemaManager.class)).deleteSchema();
+    ((TestOperateElasticsearchSchemaManager) bean(SchemaManager.class)).deleteSchema();
+    ((TestTasklistElasticsearchSchemaManager) bean(SchemaManager.class)).deleteSchema();
```
Comment (houssain-barouni) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java:140]: one last thing, I think you will get `ClassCastException` here
You need to use `bean(io.camunda.tasklist.schema.manager.SchemaManager.class)`

Comment (nathansandi) [reply qa/util/src/main/java/io/camunda/qa/util/cluster/TestStandaloneCamunda.java]: True :) for some reason my IntelliJ has not threw an error over this. But was running from the cache, which is strange. 

After clean and install I could validate this, and refactor :) Well noted.

**Review by houssain-barouni — APPROVED** [review]

**Review by ChrisKujawa — APPROVED** [review]
